File: from-java-to-kotlin/src/main/java/pl/braintelligence/E05_FieldsProperties.java
package pl.braintelligence;

import lombok.Getter;

public class E05_FieldsProperties {

    public static void main(String[] args) {
        Person person = new Person();
        person.setName("Other Jack");

        System.out.println(person.getName());
    }

    @Getter
    public static class Person {
        private String name = "Jack";

        public void setName(String name) {
            if (name != null && !name.isEmpty()) {
                this.name = name;
            }
        }
    }

}

File: from-java-to-kotlin/src/main/java/pl/braintelligence/E05_FieldsProperties.kt
package pl.braintelligence

class Person {
    var name: String = "Jack"
        set(name) {
            if (name.isNotEmpty()) field = name
        }
}

fun main() {
    val person = Person()
    person.name = "Other Jack"

    print(person.name)
}
